src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -727,7 +727,12 @@ private BoundExpression RebindSimpleBinaryOperatorAsConverted(BoundBinaryOperato
             BoundExpression? left = null;
             while (stack.TryPop(out current))
             {
-                Debug.Assert(current.Right is BoundUnconvertedInterpolatedString);
+                var right = current.Right switch
+                {
+                    BoundUnconvertedInterpolatedString s => s,
+                    BoundBinaryOperator b => RebindSimpleBinaryOperatorAsConverted(b, diagnostics),
```

Comment:
>RebindSimpleBinaryOperatorAsConverted(b, diagnostics) [](http://example.com/codeflow?start=45&length=53)

It looks like the algorithm is not symmetrical. It applies this transformation on the right, but never on the left.

Reply:
I'm not sure what you mean. This is the recursive call for the right side.

Reply:
> I'm not sure what you mean. This is the recursive call for the right side.

The recursive call will ```TryBindUnconvertedBinaryOperatorToDefaultInterpolatedStringHandler```. So subexpressions on the right might potentially use default handler, but similar sub expressions on the left might wouldn't.

Reply:
Ah, I see what you're saying there. Yes, that's a good catch (and I will add a test that would bind differently in that case).